frame/clearing-house/src/lib.rs

```diff
@@ -1099,72 +1149,66 @@ pub mod pallet {
 	// Helper functions - core functionality
 	impl<T: Config> Pallet<T> {
 		fn summarize_account_state(
+			account_id: &T::AccountId,
 			positions: BoundedVec<Position<T>, T::MaxPositions>,
 		) -> Result<AccountSummary<T>, DispatchError> {
-			let mut summary = AccountSummary::<T>::default();
+			let collateral = Self::get_collateral(account_id).unwrap_or_else(Zero::zero);
+
+			let mut summary = AccountSummary::<T>::new(collateral)?;
 			for position in positions {
 				let market = Self::try_get_market(&position.market_id)?;
 				if let Some(direction) = position.direction() {
 					// should always succeed
```

Comment:
is there a reason you have this in an if let statement if it will always pass? Is it to avoid a .unwrap()?

Reply:
Its fine either way I am just curious

Reply:
Yeah, I'm just trying to avoid a `.unwrap()`